### PR TITLE
Feature full authentication send

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://github.com/golemfactory/concent/issues/209.

Recreated from https://github.com/golemfactory/concent/pull/249 as it was automatically closed by github by mistake.

### CODE REVIEW

**Comment by rwrzesien** [comment]
@dybi All fixes applied, please review.

File: concent_api/core/tests/test_auth_accept_or_reject_results.py [hunk 1]
```diff
@@ -708,25 +773,64 @@ def test_requestor_sends_subtask_results_rejected_and_concent_should_return_it_t
...
+            )
+        )
+
+        with mock.patch('core.message_handlers.base.is_provider_account_status_positive', _get_provider_account_status_true_mock):
+            with freeze_time("2018-02-05 10:00:44"):
+                response = self.client.post(
+                    reverse('core:send'),
+                    data                            = serialized_force_subtask_results_response,
+                    content_type                    = 'application/octet-stream',
+                    HTTP_CONCENT_CLIENT_PUBLIC_KEY  = self._get_encoded_provider_public_key(),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_accept_or_reject_results.py]: Do we still need this in header?

File: concent_api/core/tests/test_auth_get_task_result.py [hunk 2]
```diff
@@ -101,72 +101,78 @@ def test_requestor_forces_get_task_result_and_concent_immediately_sends_rejectio
...
+        )
 
-        message_from_concent = load(response.content, self.DIFFERENT_REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY, check_time = False)
+        with mock.patch('core.transfer_operations.request_upload_status', request_upload_status_false_mock):
+            with freeze_time("2017-12-01 11:00:05"):
+                response = self.client.post(
+                    reverse('core:send'),
+                    data=serialized_force_get_task_result,
+                    content_type='application/octet-stream',
+                    HTTP_CONCENT_CLIENT_PUBLIC_KEY=self._get_encoded_requestor_different_public_key(),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_get_task_result.py]: Same here

File: concent_api/core/tests/test_auth_get_task_result.py [hunk 3]
```diff
@@ -101,72 +101,78 @@ def test_requestor_forces_get_task_result_and_concent_immediately_sends_rejectio
...
+            with freeze_time("2017-12-01 11:00:05"):
                 response = self.client.post(
                     reverse('core:send'),
-                    data                                = serialized_force_get_task_result,
-                    content_type                        = 'application/octet-stream',
-                    HTTP_CONCENT_CLIENT_PUBLIC_KEY      = self._get_encoded_provider_public_key(),
-                    HTTP_CONCENT_OTHER_PARTY_PUBLIC_KEY = self._get_encoded_requestor_public_key(),
+                    data=serialized_force_get_task_result,
+                    content_type='application/octet-stream',
+                    HTTP_CONCENT_CLIENT_PUBLIC_KEY=self._get_encoded_requestor_different_public_key(),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_get_task_result.py]: Same here

File: concent_api/core/tests/test_auth_report_computed_task.py [hunk 4]
```diff
@@ -166,22 +172,90 @@ def test_provider_forces_computed_task_report_and_requestor_sends_acknowledgemen
...
+                )
+            )
+        serialized_ack_report_computed_task = dump(ack_report_computed_task, self.REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+
+        with freeze_time("2017-12-01 11:00:05"):
+            response = self.client.post(
+                reverse('core:send'),
+                data                           = serialized_ack_report_computed_task,
+                content_type                   = 'application/octet-stream',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_report_computed_task.py]: Same here

File: concent_api/core/tests/test_auth_report_computed_task.py [hunk 5]
```diff
@@ -166,22 +172,90 @@ def test_provider_forces_computed_task_report_and_requestor_sends_acknowledgemen
...
+            )
+        serialized_ack_report_computed_task = dump(ack_report_computed_task, self.REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
 
         with freeze_time("2017-12-01 11:00:05"):
             response = self.client.post(
                 reverse('core:send'),
                 data                           = serialized_ack_report_computed_task,
                 content_type                   = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.PROVIDER_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_report_computed_task.py]: Same here

File: concent_api/core/tests/test_auth_report_computed_task.py [hunk 6]
```diff
@@ -420,20 +502,96 @@ def test_provider_forces_computed_task_report_and_requestor_sends_rejection_due_
                 reverse('core:send'),
                 data                           = serialized_reject_report_computed_task,
                 content_type                   = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.DIFFERENT_REQUESTOR_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_report_computed_task.py]: Same here

File: concent_api/core/tests/test_auth_report_computed_task.py [hunk 7]
```diff
@@ -420,20 +502,96 @@ def test_provider_forces_computed_task_report_and_requestor_sends_rejection_due_
...
+            )
+        serialized_reject_report_computed_task = dump(reject_report_computed_task, self.REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
 
         with freeze_time("2017-12-01 11:00:05"):
             response = self.client.post(
                 reverse('core:send'),
                 data                           = serialized_reject_report_computed_task,
                 content_type                   = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.PROVIDER_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_report_computed_task.py]: Same here

File: concent_api/core/tests/test_auth_report_computed_task.py [hunk 8]
```diff
@@ -420,20 +502,96 @@ def test_provider_forces_computed_task_report_and_requestor_sends_rejection_due_
...
+                reason              = message.RejectReportComputedTask.REASON.TaskTimeLimitExceeded
+            )
+        serialized_reject_report_computed_task = dump(reject_report_computed_task, self.REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+
+        with freeze_time("2017-12-01 11:00:05"):
+            response = self.client.post(
+                reverse('core:send'),
+                data                           = serialized_reject_report_computed_task,
+                content_type                   = 'application/octet-stream',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_report_computed_task.py]: Same here

File: concent_api/core/tests/test_auth_report_computed_task.py [hunk 9]
```diff
@@ -680,20 +904,57 @@ def test_provider_forces_computed_task_report_and_requestor_sends_rejection_due_
                 reverse('core:send'),
                 data                           = serialized_reject_report_computed_task,
                 content_type                   = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.DIFFERENT_REQUESTOR_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
```
Comment (kbeker) [inline concent_api/core/tests/test_auth_report_computed_task.py]: Same here

**Pull request closed without merging**